internal/resource/statefulset.go

```diff
@@ -499,8 +521,19 @@ func (builder *StatefulSetBuilder) podTemplateSpec(previousPodAnnotations map[st
 					SecurityContext: &corev1.SecurityContext{
 						RunAsUser: pointer.Int64Ptr(0),
 						Capabilities: &corev1.Capabilities{
-							Drop: []corev1.Capability{"ALL"},
-							Add:  []corev1.Capability{"CHOWN", "FOWNER"},
+							Drop: []corev1.Capability{
+								"FSETID",
+								"KILL",
+								"SETGID",
+								"SETUID",
+								"SETPCAP",
+								"NET_BIND_SERVICE",
+								"NET_RAW",
+								"SYS_CHROOT",
+								"MKNOD",
+								"AUDIT_WRITE",
+								"SETFCAP",
+							},
```

Comment:
@coro proposed in https://github.com/rabbitmq/cluster-operator/pull/303#discussion_r485026855 to drop all and only allow the needed capabilities.
Allow listing rather than deny listing is a better security standard.

IIRC the corresponding bug in the ESXi runtime will be fixed soon.
@Zerpet was discovering that issue and was in touch with that team.

Reply:
As I said in standup, I am aware of this fix: https://gitlab.eng.vmware.com/core-build/spherelet-agent/commit/31b0f133d2fca969b4e561d8f05a26b1f08dd498. However, It's unclear to me when this fix can be released and how often does user upgrade while using these environments with spherelet. As long as we are dropping all known unneeded capabilities, I don't see much difference of security protection between the two different approaches. In addition to that, this is only the security context for our initContainer, which is ephemeral and not long running. IMO we should merge this update to make sure that our operator work with spherelet as soon as possible. Thoughts? @mkuratczyk  @coro

Reply:
Oh right, I thought the bug mentioned in standup was a different one. My bad.
In that case, let's merge this PR as is and revert this change in the future once the fix is rolled out?

Reply:
Feel free to add a comment like `// Remove default capabilities allowed by Docker except for CHOWN and FOWNER`. Otherwise, it's hard to read what capabilities the container will run with.